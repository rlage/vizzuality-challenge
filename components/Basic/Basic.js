import React from 'react';

const Basic = ({ data }) => {
  console.log(data);
  return(
    <div> Basic </div>
  )
}

export default Basic;