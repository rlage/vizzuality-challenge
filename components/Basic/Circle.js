import React from 'react';

const Circle = ({ x, y, color, radius }) => {
  return(
    <circle data-testid='circle' cx={x} cy={y} fill={color} r={radius} ></circle>
  )
}

export default Circle;