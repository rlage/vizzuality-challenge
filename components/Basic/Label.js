import React from 'react';
import styles from './Basic.module.css'

const Label = ({ x, y, text }) => {
  return(
    <text className={styles.text} x={x} y={y}> {text} </text>
  )
}

export default Label;