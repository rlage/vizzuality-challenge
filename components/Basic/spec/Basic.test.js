import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import Basic from '../Basic';

const basicData = {
  "id": "2a022fc2-bfed-477e-9a16-75a1fb594620",
  "name": "2017 African and Asian Conflict and Protest Events",
  "type": "basic",
  "items": [
    {
      "color": "#cab2d6",
      "name": "Riots/Protests"
    },
    {
      "color": "#8dd3c7",
      "name": "Violence against civilians"
    },
    {
      "color": "#b15928",
      "name": "Battle-No change of territory"
    },
    {
      "color": "#e31a1c",
      "name": "Remote violence"
    },
    {
      "color": "#fb9a99",
      "name": "Strategic development"
    },
    {
      "color": "#33a02c",
      "name": "Battle-Government regains territory"
    },
    {
      "color": "#b2df8a",
      "name": "Battle-Non-state actor overtakes territory"
    },
    {
      "color": "#1f78b4",
      "name": "Non-violent transfer of territory"
    },
    {
      "color": "#a6cee3",
      "name": "Headquarters or base established"
    }
  ],
  "description": "<p> Two faith deadshot al! Crane ragdoll collector night, batcave tumbler master cobblepot lillian canary black. Barrow dick boomerang chimera ra doom bat ghul batcave barbara ali scorn prey. Todd maxie green tim hood collector blue abbott birds deadshot kane rose catwoman. Joker gearhead phantom zucco katana cluemaster strange dick batcave abbott copperhead echo. Kane rhino echo falcone joker maxie, dick aquaman mister catwoman. Quinn firefly tim lazarus. Riddler firefly poison riddler ra hood; america shadow. </p> <p> Drake todd temblor arkham metallo mask. Hugo batarang justice of thorne ivy catwoman freeze caird alcor bat thorne. Robin abdullah mugsy ali supergirl prey lantern. Cypher young batarang falcone, master manor bruce. Rupert al owl gearhead dick green rupert hush. Vicki lucius abbott robin shiva beetle aquaman lucius jason mad scarecrow riddler. League edward gordon hush mask batcave ragman gotham batmobile lillian. Fox solomon dent solomon abdullah atkins pennyworth alcor freeze hammer nocturna wing helena. Firefly lazarus harley society outsider the. Shiva faith owl poison mad maroni toymaker phantom alcor ra clayface. Of outsider calendar, zeus spoiler a ventriloquist beetle? </p> <p> Doom blake hood gleeson nyssa temblor calendar cain alfred ventriloquist abbott! Hood mask america shade fox. Zatanna master america croc blue hangman quinn killer poison al. Croc oswald riddler toymaker abdullah edward ivy! Blink azrael shade harvey thorne toymaker damian. Batman metallo firefly czonk chase mugsy thorne beetle bane. Tally mask aquaman lucius tally lillian. Scarecrow aiko ivy bruce chimera penguin fright! Spoiler wayne mad falcone. Society jester lynx batarang black diamond amanda croc bruce freeze. Vale aquaman. </p> <p> Abdullah abattoir ali lynx, spectre lex checkmate czonk martha green lucius! Swamp canary joker bane collector joe todd charlatan lynx bullock croc edward. Kane flash lynx hangman ivy snake a ragdoll spoiler robin caird. Swamp superman kane rumor bartok diamond atomic joker, league atomic tumbler firebug bullock. Blink copperhead aiko joe america czonk calendar zeus croc killer! Amanda amanda cluemaster smoke society knight al? Hammer scorn mugsy montoya al nyssa firefly kyle faith fairchild montoya! Lucius deathstroke diamond drake zucco. Gordon ra carmine hood lillian clock gargoyle mask penguin? Gargoyle tumbler wing batcave mugsy moth ali falcone carmine barrow. Superman justice chase doom gotham atomic lillian, luthor amanda raatko beetle pit batmobile. Solomon aiko vicki luthor vale damian ivy hangman solomon mugsy. Nyssa ra oracle rose ragdoll ali outsider fox raatko young. Clench arkham lex checkmate zatanna maxie. Arrow. </p> <p> Batgirl lucius rupert harley ali pit oracle al hugo red kobra. Calendar al crane moth mask atkins barrow scorn vale nocturna. Alcor mad wayne katana lucius gargoyle shrike tally. Of katana lillian thorne. Bane outsider lucius hatter shadow dent checkmate spectre. Ghul, cobblepot joe lantern katana? Black ragman two ra barbara, batmobile gearhead supergirl czonk amanda. Cain society two calendar fox blake. Grayson diamond league lynx crane amanda canary pit dick owl creeper! Young justice, batmobile harley nocturna! Batcave green firefly atomic elongated tally. Katana montoya shrike tally riddler edward batgirl tally prey chill amanda mad! </p>"
}

describe('<Basic /> spec', () => {
  it('renders component', () => {
    const wrapper = render(
      <Basic
        data={basicData}
      />);
    expect(wrapper).toMatchSnapshot();
  });
  it('renders all circles', () => {
    const wrapper = render(
      <Basic
        data={basicData}
      />);
    expect(wrapper.getAllByTestId('circle').length).toBe(basicData.items.length);
  });
});
