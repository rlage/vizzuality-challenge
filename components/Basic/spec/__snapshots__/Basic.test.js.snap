// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Basic /> spec renders component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <g>
        <g
          transform="translate(0 10)"
        >
          <circle
            cx="0"
            cy="0"
            data-testid="circle"
            fill="#cab2d6"
            r="6"
          />
          <text
            x="11"
            y="3"
          >
             
            Riots/Protests
             
          </text>
        </g>
        <g
          transform="translate(0 29)"
        >
          <circle
            cx="0"
            cy="0"
            data-testid="circle"
            fill="#8dd3c7"
            r="6"
          />
          <text
            x="11"
            y="3"
          >
             
            Violence against civilians
             
          </text>
        </g>
        <g
          transform="translate(0 48)"
        >
          <circle
            cx="0"
            cy="0"
            data-testid="circle"
            fill="#b15928"
            r="6"
          />
          <text
            x="11"
            y="3"
          >
             
            Battle-No change of territory
             
          </text>
        </g>
        <g
          transform="translate(0 67)"
        >
          <circle
            cx="0"
            cy="0"
            data-testid="circle"
            fill="#e31a1c"
            r="6"
          />
          <text
            x="11"
            y="3"
          >
             
            Remote violence
             
          </text>
        </g>
        <g
          transform="translate(0 86)"
        >
          <circle
            cx="0"
            cy="0"
            data-testid="circle"
            fill="#fb9a99"
            r="6"
          />
          <text
            x="11"
            y="3"
          >
             
            Strategic development
             
          </text>
        </g>
        <g
          transform="translate(0 105)"
        >
          <circle
            cx="0"
            cy="0"
            data-testid="circle"
            fill="#33a02c"
            r="6"
          />
          <text
            x="11"
            y="3"
          >
             
            Battle-Government regains territory
             
          </text>
        </g>
        <g
          transform="translate(0 124)"
        >
          <circle
            cx="0"
            cy="0"
            data-testid="circle"
            fill="#b2df8a"
            r="6"
          />
          <text
            x="11"
            y="3"
          >
             
            Battle-Non-state actor overtakes territory
             
          </text>
        </g>
        <g
          transform="translate(0 143)"
        >
          <circle
            cx="0"
            cy="0"
            data-testid="circle"
            fill="#1f78b4"
            r="6"
          />
          <text
            x="11"
            y="3"
          >
             
            Non-violent transfer of territory
             
          </text>
        </g>
        <g
          transform="translate(0 162)"
        >
          <circle
            cx="0"
            cy="0"
            data-testid="circle"
            fill="#a6cee3"
            r="6"
          />
          <text
            x="11"
            y="3"
          >
             
            Headquarters or base established
             
          </text>
        </g>
      </g>
    </div>
  </body>,
  "container": <div>
    <g>
      <g
        transform="translate(0 10)"
      >
        <circle
          cx="0"
          cy="0"
          data-testid="circle"
          fill="#cab2d6"
          r="6"
        />
        <text
          x="11"
          y="3"
        >
           
          Riots/Protests
           
        </text>
      </g>
      <g
        transform="translate(0 29)"
      >
        <circle
          cx="0"
          cy="0"
          data-testid="circle"
          fill="#8dd3c7"
          r="6"
        />
        <text
          x="11"
          y="3"
        >
           
          Violence against civilians
           
        </text>
      </g>
      <g
        transform="translate(0 48)"
      >
        <circle
          cx="0"
          cy="0"
          data-testid="circle"
          fill="#b15928"
          r="6"
        />
        <text
          x="11"
          y="3"
        >
           
          Battle-No change of territory
           
        </text>
      </g>
      <g
        transform="translate(0 67)"
      >
        <circle
          cx="0"
          cy="0"
          data-testid="circle"
          fill="#e31a1c"
          r="6"
        />
        <text
          x="11"
          y="3"
        >
           
          Remote violence
           
        </text>
      </g>
      <g
        transform="translate(0 86)"
      >
        <circle
          cx="0"
          cy="0"
          data-testid="circle"
          fill="#fb9a99"
          r="6"
        />
        <text
          x="11"
          y="3"
        >
           
          Strategic development
           
        </text>
      </g>
      <g
        transform="translate(0 105)"
      >
        <circle
          cx="0"
          cy="0"
          data-testid="circle"
          fill="#33a02c"
          r="6"
        />
        <text
          x="11"
          y="3"
        >
           
          Battle-Government regains territory
           
        </text>
      </g>
      <g
        transform="translate(0 124)"
      >
        <circle
          cx="0"
          cy="0"
          data-testid="circle"
          fill="#b2df8a"
          r="6"
        />
        <text
          x="11"
          y="3"
        >
           
          Battle-Non-state actor overtakes territory
           
        </text>
      </g>
      <g
        transform="translate(0 143)"
      >
        <circle
          cx="0"
          cy="0"
          data-testid="circle"
          fill="#1f78b4"
          r="6"
        />
        <text
          x="11"
          y="3"
        >
           
          Non-violent transfer of territory
           
        </text>
      </g>
      <g
        transform="translate(0 162)"
      >
        <circle
          cx="0"
          cy="0"
          data-testid="circle"
          fill="#a6cee3"
          r="6"
        />
        <text
          x="11"
          y="3"
        >
           
          Headquarters or base established
           
        </text>
      </g>
    </g>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
